frontend/src/features/resources/ManageResourcesPage.jsx

```diff
@@ -0,0 +1,559 @@
+                          {f.groupName} / {f.childName}
+                        </td>
+                        <td className="text-right text-xs">
+                          {formatBytes(f.size)}
+                        </td>
+                        <td className="text-right">
+                          <button
+                            className="btn btn-ghost btn-xs text-error"
+                            onClick={() => removeFile(f.id)}
+                          >
+                            Delete
+                          </button>
+                        </td>
+                      </tr>
+                    ))}
+                  </tbody>
+                </table>
+              </div>
+            )}
+
+            <div className="divider my-2" />
+
+            <Preview file={selectedFile} />
+          </div>
+        </div>
+      </div>
+    </ScreenContainer>
+  );
+}
+
+// ===== Small components =====
+function UploadButton({ onFiles, multiple = true, className = "" }) {
+  const inputRef = useRef(null);
+  return (
+    <>
+      <input
+        ref={inputRef}
+        type="file"
+        multiple={multiple}
+        className="hidden"
+        onChange={(e) => {
+          const files = Array.from(e.target.files || []);
+          if (files.length) onFiles(files);
+          e.target.value = ""; // allow re-select
+        }}
+      />
+      <button
+        className={`btn btn-ghost btn-xs ${className}`}
+        onClick={() => inputRef.current?.click()}
+        title="Add files"
+      >
+        ＋
+      </button>
+    </>
+  );
+}
+
+function AddGroup({ onAdd }) {
+  const [open, setOpen] = useState(false);
+  const [name, setName] = useState("");
+  return (
+    <div className="dropdown dropdown-end">
+      <button className="btn btn-sm" onClick={() => setOpen((v) => !v)}>
+        New group
+      </button>
+      {open && (
+        <div className="dropdown-content z-[1] bg-base-100 rounded-box p-3 w-64 shadow">
+          <label className="form-control w-full">
+            <span className="label-text">Group name</span>
+            <input
+              className="input input-bordered input-sm"
+              value={name}
+              onChange={(e) => setName(e.target.value)}
+            />
+          </label>
+          <div className="mt-3 flex justify-end gap-2">
+            <button
+              className="btn btn-ghost btn-sm"
+              onClick={() => setOpen(false)}
+            >
+              Cancel
+            </button>
+            <button
+              className="btn btn-primary btn-sm"
+              onClick={() => {
+                if (!name.trim()) return;
+                onAdd(name.trim());
+                setName("");
+                setOpen(false);
+              }}
+            >
+              Add
+            </button>
+          </div>
+        </div>
+      )}
+    </div>
+  );
+}
+
+function AddChild({ onAdd }) {
```

Comment:
generally if you've got decently sized components, you'd put them in seperate files (which i would do here for `AddGroup` and `AddChild`).